cyto_dl/api/data.py

```diff
@@ -0,0 +1,40 @@
+# goal: allow the API client to interact with CytoDLModel without
+# worrying about the actual values being used in config files
+from enum import Enum
+
+import skimage
+
+
+class ExperimentType(Enum):
+    GAN = "gan"
+    INSTANCE_SEG = "instance_seg"
+    LABEL_FREE = "labelfree"
+    SEGMENTATION_PLUGIN = "segmentation_plugin"
+    SEGMENTATION = "segmentation"
+
+
+class HardwareType(Enum):
+    CPU = "cpu"
+    GPU = "gpu"
+    # other hardware types available, but require more complicated config
+
+
+class PatchSize(Enum):
+    """Patch size for training, and their respective patch shapes."""
+
+    SMALL = [16, 32, 32]
+    MEDIUM = [16, 64, 64]
+    LARGE = [16, 128, 128]
+
+
+# importing skimage takes a while.
+# could speed it up by hardcoding this enum, but would have to update
+# if skimage adds/deletes threshold filters
+
+AutoThresholdMethod = Enum(
+    "AutoThresholdMethod",
+    {
+        func_name.split("threshold_")[-1].upper(): func_name
+        for func_name in filter(lambda x: x.startswith("threshold_"), dir(skimage.filters))
```

Comment:
Is `skimage` pinned?  I imagine that `skimage.filters` could change and break assumpions in this code

Reply:
I dont think i understand how `AutoThresholdMethod` is used - is it exercised by `model.load_default_experiment` (and called by the new tests)?

Reply:
Ok, i get it after seeing the example in your [doc](https://alleninstitute-my.sharepoint.com/:w:/g/personal/daniel_saelid_alleninstitute_org/EaRGLGO2zgRDmAC8nX5ChocBcxePXOTnlcHQxODIbaVA8g?e=99ASMS).  How about including a docstring with that usage example?